### DISCUSSION TITLE
How can I get the message of x,y or the velocity of x,y axis with IMU

### DISCUSSION THREAD

**Discussion by Ericzhaoforu** [bitcraze/discussions#723] [Q&A]
Hi!
     I've been trying to develop a position-control method using cflib, but I can't get information about x,y, which is important to my controller.
So how can I get them without using decks? I've been trying to switch the state estimator to EKF to get  them, but is it accuracy?And when I log the pram of z, it always show 400+, but when the crazyflie takes off it turns to around zero.

**Comment by krichardsson** [comment]
The short answer is unfortunately that you can't. Your approach is right, the kalman estimator keeps track of the position, but the problem is that without any deck, it is only using the IMU (accelerometer and gyro) as an input, and the IMU is too noisy for any decent positioning.
You could reset the kalman estimator just before take off, but it will probably only give you reasonable position information for a second or so.

> And when I log the pram of z, it always show 400+, but when the crazyflie takes off it turns to around zero.

I'm not exactly sure why you get this behavior.
